liquidctl/driver/kraken_three_x.py

```diff
@@ -0,0 +1,327 @@
+            """
+            num_light_channels = msg[14]  # the 15th byte (index 14) is # of light channels
+            accessories_per_channel = 6  # each lighting channel supports up to 6 accessories
+            light_accessory_index = 15  # offset in msg of info about first light accessory
+            for light_channel in range(num_light_channels):
+                for accessory_num in range(accessories_per_channel):
+                    accessory_id = msg[light_accessory_index]
+                    light_accessory_index += 1
+                    if accessory_id != 0:
+                        status.append(('LED {} accessory {}'.format(light_channel + 1, accessory_num + 1),
+                                       _ACCESSORY_NAMES.get(accessory_id, 'Unknown'), ''))
+
+        self._read_until({b'\x11\x01': parse_firm_info, b'\x21\x03': parse_led_info})
+        self.device.release()
+        return sorted(status)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        msg = self._read()
+        return [
+            ('Liquid temperature', msg[15] + msg[14] / 10, '°C'),
+            ('Pump speed', msg[18] << 8 | msg[17], 'rpm'),
+            ('Pump duty', msg[19], '%'),
+        ]
+
+    def set_color(self, channel, mode, colors, speed='normal', **kwargs):
+        """Set the color mode for a specific channel."""
+        cid = _COLOR_CHANNELS[channel]
+        _, _, _, mincolors, maxcolors = _COLOR_MODES[mode]
+        colors = [[g, r, b] for [r, g, b] in colors]
+        if len(colors) < mincolors:
+            raise ValueError('Not enough colors for mode={}, at least {} required'.format(mode, mincolors))
+        elif maxcolors == 0:
+            if colors:
+                LOGGER.warning('too many colors for mode=%s, none needed', mode)
+            colors = [[0, 0, 0]]  # discard the input but ensure at least one step
+        elif len(colors) > maxcolors:
+            LOGGER.warning('too many colors for mode=%s, dropping to %i', mode, maxcolors)
+            colors = colors[:maxcolors]
+        sval = _ANIMATION_SPEEDS[speed]
+        self._write_colors(cid, mode, colors, sval)
+        self.device.release()
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to use a speed profile."""
+        if channel != 'pump':
+            assert False, 'kraken X3 devices only support changing pump speeds'
+        header = [0x72, 0x01, 0x00, 0x00]
+        norm = normalize_profile(profile, _CRITICAL_TEMPERATURE)
+        interp = [(interpolate_profile(norm, t)) for t in range(20, 60)]
+        LOGGER.debug('setting pump curve: %s', [(num+20, duty) for (num, duty) in enumerate(interp)])
+        raise NotImplementedError()  # precaution
+        self._write(header + interp)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
```

Comment:
You can move & delegate everything to `set_speed_profile`:

```py
    def set_fixed_speed(self, channel, duty, **kwargs):
        """Set channel to a fixed speed."""
        self.set_speed_profile(channel, [(0, duty), (100, 100)])
```

Reply:
```python
    def set_fixed_speed(self, channel, duty, **kwargs):
        """Set channel to a fixed speed duty."""
        duty = clamp(duty, 20, 100)
        self.set_speed_profile(channel, [(0, duty), (100, 100)])
```

Unfortunately this does not work as expected. Only setting it to 100 works as it should. 

`liquidctl set pump speed 20`
`[DEBUG] liquidctl.driver.kraken_three_x: write 72 01 00 00 2f 30 32 33 35 36 37 39 3a 3b 3d 3e 3f 41 42 43 45 46 48 49 4a 4c 4d 4e 50 51 52 54 55 56 58 59 5b 5c 5d 5f 60 61 63 64 (and 20 padding bytes)`

`liquidctl set pump speed 60`
`[DEBUG] liquidctl.driver.kraken_three_x: write 72 01 00 00 4a 4a 4b 4c 4c 4d 4e 4e 4f 50 50 51 52 52 53 54 54 55 56 56 57 58 58 59 5a 5b 5b 5c 5d 5d 5e 5f 5f 60 61 61 62 63 63 64 (and 20 padding bytes)`

`liquidctl set pump speed 100`
`[DEBUG] liquidctl.driver.kraken_three_x: write 72 01 00 00 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 64 (and 20 padding bytes)`

And it should be for example: 
`liquidctl set pump speed 60`
`[DEBUG] liquidctl.driver.kraken_three_x: write 72 01 00 00 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c 3c (and 20 padding bytes)`

Reply:
Oh, I have a feeling I made the exact same mistake before.

The following should work then (from [`KrakenTwoDriver`](https://github.com/jonasmalacofilho/liquidctl/blob/1f370d2507fc5a7f95efb465764fc46365ce9587/liquidctl/driver/kraken_two.py#L231)):

```py
def set_fixed_speed(self, channel, duty, **kwargs):
    """Set channel to a fixed speed."""
    self.set_speed_profile(channel, [(0, duty), (59, duty), (60, 100), (100, 100)])
```

Also, clamping can be left to `set_speed_profile` too.

Reply:
The problem was that I forgot that the critical temperature is 59°C in this driver, and thus `normalize_profile` would insert an additional (59°C, 100%) point in the profile, causing the slope.